server/services/starUpgrade.js

```diff
@@ -341,9 +334,22 @@ module.exports = class StarUpgradeService extends EventEmitter {
                 return {
                     star: s,
                     terraformedResources,
-                    infrastructureCost: calculateCostFunction(game, expenseConfig, s.infrastructure[infrastructureType], terraformedResources)
+                    infrastructureCost: calculateCostFunction(game, expenseConfig, s.infrastructure[infrastructureType], terraformedResources),
+                    upgradeFunction: upgradeFunction(game, player, s._id, false),
```

Comment:
Rename to `upgraded` or something, `upgradeFunction` suggests that it is actually a function and not a numeric value.